### DISCUSSION TITLE
Can systemrdl-compiler write systemrdl?

### DISCUSSION THREAD

**Discussion by edcote** [SystemRDL/systemrdl-compiler#143] [Q&A] [answered]
Assuming I create a register map directly using Python. Is there a built in method to write out RDL?

Does it require a custom RDLWalker?

**Comment by amykyta3** [comment]
Sorry I meant to reply to this.

A way to write out RDL is coming soon!

It wont be 100% symmetrical in feature-set, but it should be enough for users that want to translate from things like IP-XACT, or their own custom formats into SystemRDL

**Comment by amykyta3** [comment, marked as answer]
I have just implemented this feature in the PeakRDL command line tool. See: https://peakrdl.readthedocs.io/en/latest/gallery.html#convert-other-formats-to-systemrdl

Command-line:
```bash
peakrdl systemrdl input.xml -o output.rdl
```

Or if you want to implement in your existing tool, see [PeakRDL-systemrdl](https://github.com/SystemRDL/PeakRDL-systemrdl)
